fiftyone/core/dataset.py

```
@@ -4498,32 +4498,65 @@ def save_view(
-            slug=slug,
-            description=description,
-            color=color,
-            view_stages=[
+        # Check if we're updating an existing view
+        existing_view_doc = None
+        if overwrite:
+            try:
+                # Targeted reload to get fresh view docs
+                self._doc.reload("saved_views")
+                existing_view_doc = self._get_saved_view_doc(name)
+            except ValueError as e:
```

Ruff (0.12.2)

Local variable `e` is assigned to but never used

Remove assignment to unused variable `e`

(F841)